src/analysis/souffle/tests/test_taint_with_auth_logic.dl

```diff
@@ -0,0 +1,101 @@
+//-----------------------------------------------------------------------------
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//-----------------------------------------------------------------------------
+
+#include "../taint_with_auth_logic.dl"
+#include "../flat_graph_ir.dl"
+#include "../edges_from_graph.dl"
+
+// This is a datalog file that is meant to test taint_with_auth_logic.dl. It 
+// should eventually be replaced by or otherwise used as a part of an
+// automated test.
+
+//-----------------------------------------------------------------------------
+// Program Manifests
+//-----------------------------------------------------------------------------
+
+// Module 1
```

Comment:
Don't mean to be a pain, but can we give some meaningful names to modules, inputs, and outputs? I find it hard to follow how things work.

Reply:
Will do... in progress..

Reply:
OK made the test more realistic and with better names. Also the comment spells out the downgrade/upgrade sequence in more detail.